libraries/AP_HAL_ChibiOS/hwdef/CUAV-7-Nano/hwdef.dat

```diff
@@ -0,0 +1,305 @@
+
+# SPI5 - FRAM
+PF7  SPI5_SCK  SPI5
+PH7  SPI5_MISO SPI5
+PF11 SPI5_MOSI SPI5
+PG7  FRAM_CS CS
+
+# SPI devices 
+SPIDEV bmi088_g SPI2 DEVID1  BMI088_G_CS    MODE3 10*MHZ  10*MHZ
+SPIDEV bmi088_a SPI2 DEVID2  BMI088_A_CS    MODE3 10*MHZ  10*MHZ
+SPIDEV iim42652 SPI1 DEVID1  IIM42652_CS    MODE3  2*MHZ   8*MHZ
+SPIDEV ramtron  SPI5 DEVID1  FRAM_CS        MODE3  8*MHZ   8*MHZ
+SPIDEV bmp581   SPI4 DEVID1  BMP581_CS      MODE3 7.5*MHZ 12*MHZ
+
+# PWM output pins
+PI0  TIM5_CH4  TIM5  PWM(1)  GPIO(50)
+PH12 TIM5_CH3  TIM5  PWM(2)  GPIO(51)
+PH11 TIM5_CH2  TIM5  PWM(3)  GPIO(52)
+PH10 TIM5_CH1  TIM5  PWM(4)  GPIO(53)
+PD13 TIM4_CH2  TIM4  PWM(5)  GPIO(54)
+PD14 TIM4_CH3  TIM4  PWM(6)  GPIO(55)
+PE11 TIM1_CH2  TIM1  PWM(7)  GPIO(56)
+PE9  TIM1_CH1  TIM1  PWM(8)  GPIO(57)
+PI6  TIM8_CH2  TIM8  PWM(9)  GPIO(58)
+PI7  TIM8_CH3  TIM8  PWM(10) GPIO(59)
+PI5  TIM8_CH1  TIM8  PWM(11) GPIO(60)
+PE6  TIM15_CH2 TIM15 PWM(12) GPIO(61)
+
+# we need to disable DMA on the last 2 FMU channels
+# as timer 12 doesn't have a TIMn_UP DMA option
+PH6  TIM12_CH1 TIM12 PWM(13) GPIO(62) NODMA
+PH9  TIM12_CH2 TIM12 PWM(14) GPIO(63) NODMA
+
+# CAN bus
+PD0  CAN1_RX CAN1
+PD1  CAN1_TX CAN1
+
+PB12 CAN2_RX CAN2
+PB13 CAN2_TX CAN2
+
+# control for silent (no output) for CAN
+PE2 GPIO_CAN1_SILENT OUTPUT PUSHPULL SPEED_LOW LOW GPIO(70)
+PI8 GPIO_CAN2_SILENT OUTPUT PUSHPULL SPEED_LOW LOW GPIO(71)
+
+# I2C buses
+
+# I2C1, GPS+MAG
+PB9 I2C1_SDA I2C1
+PB8 I2C1_SCL I2C1
+
+# I2C2, GPS2+MAG
+PF1 I2C2_SCL I2C2
+PF0 I2C2_SDA I2C2
+
+# I2C3, IST8310
+PA8 I2C3_SCL I2C3
+PH8 I2C3_SDA I2C3
+
+# I2C4, ICM20100
+PF14 I2C4_SCL I2C4
+PF15 I2C4_SDA I2C4
+PG5  DRDY1_ICP20100 INPUT
+
+# order of I2C buses
+I2C_ORDER I2C3 I2C4 I2C1 I2C2
+define HAL_I2C_INTERNAL_MASK 3
+
+# armed indication
+PE7 nARMED OUTPUT HIGH
+
+# power enable pins
+PC13 VDD_3V3_SD_CARD_EN OUTPUT HIGH
+PI11 VDD_3V3_SENSORS1_EN OUTPUT HIGH
+PF4  VDD_3V3_SENSORS2_EN OUTPUT HIGH
+PH2  VDD_3V3_SENSORS3_EN OUTPUT HIGH
+
+# start peripheral power off, then enable after init
+# this prevents a problem with radios that use RTS for
+# bootloader hold
+PG10 VDD_5V_HIPOWER_EN OUTPUT HIGH
+PG4  VDD_5V_PERIPH_EN  OUTPUT HIGH
+
+# power sensing
+PE15 VDD_5V_PERIPH_nOC INPUT PULLUP
+PF13 VDD_5V_HIPOWER_nOC INPUT PULLUP
+
+# Pin for PWM Voltage Selection, 0 means 3.3v, 1 means 5v, 3.3v default
+PG8 PWM_VOLT_SEL OUTPUT LOW GPIO(80)
+define HAL_GPIO_PWM_VOLT_PIN 80
+define HAL_GPIO_PWM_VOLT_3v3 0
+
+# microSD support
+PD6  SDMMC2_CK SDMMC2
+PD7  SDMMC2_CMD SDMMC2
+PB14 SDMMC2_D0 SDMMC2
+PB15 SDMMC2_D1 SDMMC2
+PG11 SDMMC2_D2 SDMMC2
+PB4  SDMMC2_D3 SDMMC2
+define FATFS_HAL_DEVICE SDCD2
+
+# safety
+PD10 LED_SAFETY OUTPUT
+PF5  SAFETY_IN INPUT PULLDOWN
+
+# LEDs
+PE3  LED_RED   OUTPUT GPIO(90) LOW
+PE4  LED_GREEN OUTPUT GPIO(91) LOW
+PE5  LED_BLUE  OUTPUT GPIO(92) LOW
+
+# setup for "pixracer" RGB LEDs
+define HAL_GPIO_A_LED_PIN 90
+define HAL_GPIO_B_LED_PIN 91
+define HAL_GPIO_C_LED_PIN 92
+define HAL_GPIO_LED_ON 0
+
+define HAL_HAVE_PIXRACER_LED
```

Comment:
```suggestion
# setup for "pixracer" RGB LEDs
define AP_NOTIFY_GPIO_LED_RGB_RED_PIN 90
define AP_NOTIFY_GPIO_LED_RGB_GREEN_PIN 91
define AP_NOTIFY_GPIO_LED_RGB_BLUE_PIN 92
define HAL_GPIO_LED_ON 0

define AP_NOTIFY_GPIO_LED_RGB_ENABLED 1
```

Reply:
Okay, thanks.

Reply:
@cuav-chen2 if you make the suggested LED changes it will build... @makermare PR is not needed and is incorrect

Reply:
> @cuav-chen2 if you make the suggested LED changes it will build... @makermare PR is not needed and is incorrect

Yes, I have successfully built with these changes. I will update this commit later, thanks.